orcestra_book/plans/HALO-20240919a.md

```diff
@@ -48,3 +49,218 @@ takeoff: "2024-09-19T12:00:00Z"
+
+# Create points for flight path
+meteor = LatLon(10.65, -49.5).assign(label="meteor")
+pace_ec_overlap = LatLon(16.7792, -53.4286).assign(label="pace_ec_cross")
+
+pace_start = point_on_track(pace_track, lat=11.35).assign(label="pace_start")
+pace_end = point_on_track(pace_track, lat=pace_ec_overlap.lat).assign(label = "pace_end")
+dist5mins = halo_speed * 5 * 60
+pace_end_t5mins = pace_end.course(geod_azi(pace_end, pace_start), dist5mins)
+pace_under = point_on_track(pace_track, lat=pace_end_t5mins.lat, with_time=True).assign(label="pace_under", note="meet PACE")
+
+ec_start = point_on_track(ec_track, lat=pace_ec_overlap.lat-0.025).assign(label = "ec_start")
+ec_end = point_on_track(ec_track, lat=11.88).assign(label = "ec_end")
+dist15mins = halo_speed * 15 * 60
+ec_start_t15mins = ec_start.course(geod_azi(ec_start, ec_start), dist15mins)
+ec_under = point_on_track(ec_track, lat=ec_start_t15mins.lat).assign(label="ec_under", note="meet EarthCARE")
+
+dist_between_circles = radius * 2.0 / 3.0
+dist1 = geod_dist(airport, pace_start) + 3.0 * radius + 2.0 * dist_between_circles
+circ_1 = airport.course(geod_azi(airport, meteor), dist1).assign(label="c1")
+circ_3 = centre_from_tangent(meteor, pace_start, radius, 0).assign(label="c3")
+circ_2 = circ_1.towards(circ_3, fraction=1/2).assign(label="c2")
+p2 = circ_2.towards(circ_3, distance=radius).assign(label="p2")
+circ_4 = centre_from_tangent(ec_end, pace_end, radius, 180).assign(label="c4")
+circ_5 = circ_3.towards(airport, distance=geod_dist(circ_1, circ_2)).assign(label="c5")
+
+# Define Waypoints
+fl1, fl2, fl3 = 410, 430, 450
+waypoints = [
+    airport.assign(fl=0),
+    meteor.assign(fl=fl1),
+    IntoCircle(circ_1.assign(fl=fl2), radius,  angle=-360), # anti-clockwise (optional)
+    meteor.assign(fl=fl2),
+    p2.assign(fl=fl2),
+    IntoCircle(circ_2.assign(fl=fl2), radius,  angle=-360), # anti-clockwise (optional)
+    IntoCircle(circ_3.assign(fl=fl3), radius,  angle=-360), # ANTI-CLOCKWISE
+    pace_start.assign(fl=fl3),
+    pace_under.assign(fl=fl3),
+    pace_end.assign(fl=fl3),
+    IntoCircle(circ_4.assign(fl=fl3), radius,  angle=360), # CLOCKWISE
+    ec_start.assign(fl=fl3),
+    ec_under.assign(fl=fl3),
+    ec_end.assign(fl=fl3),
+    IntoCircle(circ_5.assign(fl=fl3), smaller_radius,  angle=-360), # anti-clockwise (optional)
+    airport.assign(fl=0),
+]
+
+# Additional Waypoints
+extra_waypoints = []
+
+# FlightPlan and print short statement
+plan = FlightPlan(path=waypoints, flight_id=flight_id, extra_waypoints=extra_waypoints, crew=crew, aircraft=aircraft)
+
+msg = f"Flight ID: {plan.flight_id}\n" \
+    + f"Take-off: {plan.takeoff_time:%H:%M %Z}\n" \
+    + f"Landing:  {plan.landing_time:%H:%M %Z}\n" \
+    + f"Duration: {plan.duration}"
+print(msg)
+```
+
+```{code-cell} ipython3
+:tags: [hide-input]
+from HALO_20240919a_plan import plot_flight_plan_satellite_forecast
+from orcestra.flightplan import plot_cwv
+
+figsize=(14, 8)
+worldfirs_json = "worldfirs.json"
```

Comment:
[discussed in person] I add a label to the legend for the ATC zones and since plot_firs doesn't return the line handles I can't use this function (yet...)